sql/expression/logic.go

```diff
@@ -137,9 +143,31 @@ var _ sql.CollationCoercible = (*Or)(nil)
 
 // NewOr creates a new Or expression.
 func NewOr(left, right sql.Expression) sql.Expression {
+	if left == nil {
+		return right
+	}
+	if right == nil {
+		return left
+	}
 	return &Or{BinaryExpression{Left: left, Right: right}}
 }
 
+// JoinOr joins several expressions with Or.
+func JoinOr(exprs ...sql.Expression) sql.Expression {
```

Comment:
Could you please add a unit test for this new exported function? I'm happy to accept this change, but we need test coverage. It should be trivial to add to the existing unit tests in `logic_test.go`.

Reply:
Added!